### PR TITLE
Adds notification for when tips are processed

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/brave/brave-browser/issues/3637
UI: https://github.com/brave/brave-ui/pull/429

The intention is that when AutoContribute is off, that we notify the user when their recurring tips have processed.

Screen Shot 2019-03-14 at 5 33 26 PM

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
1. Start Brave and enable Rewards
2. Claim a grant, turn off auto contribute
3. Make some monthly tips to different publishers
4. Wait a month (or set your reconcile stamp back to something reasonable)
5. Ensure that the notification showing that tips have been processed is in the panel after reconcile is finished.
6. Repeat the steps with auto contribute on, and ensure the notification does not show

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json [hunk 1]
```diff
@@ -376,5 +376,13 @@
   "welcomeBack": {
     "message": "Welcome Back!",
     "description": "Title message for wallet disabled panel"
+  },
+  "tipsProcessedNotification": {
+    "message": "Your monthly tips have been processed!",
+    "description": "Message for monthly tips processed notification"
+  },
+  "contributionTips": {
+    "message": "Contributions & Tips",
```
Comment (ryanml) [inline components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json]: Updated to `Contributions & Tips` per @mandar-brave

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 2]
```diff
@@ -872,6 +872,7 @@ void RewardsServiceImpl::OnReconcileComplete(ledger::Result result,
       return;
 
     FetchWalletProperties();
+    MaybeShowNotificationTipsPaid();
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/rewards_service_impl.cc]: we need to check for `ledger::REWARDS_CATEGORY` if it's recurring tip and only then call this function

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 3]
```diff
@@ -2396,6 +2397,21 @@ void RewardsServiceImpl::ShowNotificationAddFunds(bool sufficient) {
       "rewards_notification_insufficient_funds");
 }
 
+void RewardsServiceImpl::MaybeShowNotificationTipsPaid() {
+  GetAutoContribute(base::Bind(&RewardsServiceImpl::ShowNotificationTipsPaid,
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/rewards_service_impl.cc]: this should be bind once

Comment (ryanml) [reply components/brave_rewards/browser/rewards_service_impl.cc]: @NejcZdovc you just used `base::Bind` here for the exact same function, is that wrong?

Comment (NejcZdovc) [reply components/brave_rewards/browser/rewards_service_impl.cc]: yup, we should use once when we only want one response

File: components/brave_rewards/browser/rewards_service_impl.cc (hunk 2 above)
Comment (ryanml) [reply components/brave_rewards/browser/rewards_service_impl.cc]: done

File: components/brave_rewards/browser/rewards_service_impl.cc (hunk 3 above)
Comment (ryanml) [reply components/brave_rewards/browser/rewards_service_impl.cc]: done, now using BindOnce :)

**Comment by ryanml** [comment]
@NejcZdovc test plan worked fine for me, could you maybe list your steps to that path?

Screen Shot 2019-03-26 at 3 38 55 PM

**Comment by ryanml** [comment]
@NejcZdovc rebased, using `BindOnce` now in `BraveRewardsGetACEnabledFunction`

**Comment by ryanml** [comment]
@NejcZdovc C++ lint fixed

**Pull request merged**